diff --git a/arch/arm64/boot/dts/ti/k3-am642-iolan.dts b/arch/arm64/boot/dts/ti/k3-am642-iolan.dts
old mode 100755
new mode 100644
index 69eebe3f2838..10dee39136a1
--- a/arch/arm64/boot/dts/ti/k3-am642-iolan.dts
+++ b/arch/arm64/boot/dts/ti/k3-am642-iolan.dts
@@ -34,6 +34,10 @@ aliases {
 		rtc0 = &rtc_rv3028;
 	};
 
+	cpufreq {
+		compatible = "ti,am64-cpufreq";
+	};
+
 	memory@80000000 {
 		bootph-all;
 		device_type = "memory";
@@ -64,18 +68,6 @@ main_r5fss0_core0_memory_region: r5f-memory@a0100000 {
 			no-map;
 		};
 
-		main_r5fss0_core1_dma_memory_region: r5f-dma-memory@a1000000 {
-			compatible = "shared-dma-pool";
-			reg = <0x00 0xa1000000 0x00 0x100000>;
-			no-map;
-		};
-
-		main_r5fss0_core1_memory_region: r5f-memory@a1100000 {
-			compatible = "shared-dma-pool";
-			reg = <0x00 0xa1100000 0x00 0xf00000>;
-			no-map;
-		};
-
 		rtos_ipc_memory_region: ipc-memories@a5000000 {
 			reg = <0x00 0xa5000000 0x00 0x00800000>;
 			alignment = <0x1000>;
@@ -83,33 +75,41 @@ rtos_ipc_memory_region: ipc-memories@a5000000 {
 		};
 	};
 
-	vusb_main: regulator-0 {
-		/* USB MAIN INPUT 5V DC */
-		bootph-all;
+	firmware {
+		optee {
+			compatible = "linaro,optee-tz";
+			method = "smc";
+		};
+	};
+
+	/* Module Power Supply */
+	reg_vsodimm: regulator-vsodimm {
 		compatible = "regulator-fixed";
-		regulator-name = "vusb_main5v0";
-		regulator-min-microvolt = <5000000>;
-		regulator-max-microvolt = <5000000>;
-		regulator-always-on;
-		regulator-boot-on;
+		regulator-name = "+V_SODIMM";
 	};
 
-	vcc_3v3_sys: regulator-1 {
-		/* output of LP8733xx */
-		bootph-all;
+	/* Non PMIC On-module Supplies */
+	reg_3v3: regulator-3v3 {
 		compatible = "regulator-fixed";
-		regulator-name = "vcc_3v3_sys";
-		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
-		vin-supply = <&vusb_main>;
-		regulator-always-on;
-		regulator-boot-on;
+		regulator-min-microvolt = <3300000>;
+		regulator-name = "On-module +V3.3";
+		vin-supply = <&reg_vsodimm>;
 	};
 
-	firmware {
-		optee {
-			compatible = "linaro,optee-tz";
-			method = "smc";
+	a53_opp_table: opp-table {
+		compatible = "operating-points-v2";
+		opp-shared;
+
+		opp-400000000 {
+			opp-hz = /bits/ 64 <400000000>;
+			/* opp-supported-hw = <0x01 0x00>; */
+			clock-latency-ns = <600000>;
+		};
+		opp-800000000 {
+			opp-hz = /bits/ 64 <800000000>;
+			/* opp-supported-hw = <0x01 0x00>; */
+			clock-latency-ns = <600000>;
 		};
 	};
 
@@ -460,9 +460,16 @@ AM64X_IOPAD(0x0144, PIN_OUTPUT, 4) /* (Y11) PRG1_PRU1_GPO15.RGMII2_TX_CTL */
 		>;
 	};
 
+	/* PMIC_EXTINT# */
+	pinctrl_pmic_extint: main-system-extint-default-pins {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x0278, PIN_INPUT, 0) /* (C19) EXTINTn */
+		>;
+	};
+
 	main_pcie0_pins_default: main-pcie0-pins-default {
 	    pinctrl-single,pins = <
-			AM64X_IOPAD(0x0248, PIN_INPUT, 3) /* (D16) UART1_CTSn.PCIE0_CLKREQn */
+		AM64X_IOPAD(0x0248, PIN_INPUT_PULLUP, 3) /* (D16) UART1_CTSn.PCIE0_CLKREQn */
 	    >;
 	};
 
@@ -561,17 +568,32 @@ rtc_rv3028: rtc@52 {
 	tps65219: pmic@30 {
 		compatible = "ti,tps65219";
 		reg = <0x30>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_pmic_extint>;
+		interrupt-parent = <&gic500>;
+		interrupts = <GIC_SPI 224 IRQ_TYPE_LEVEL_HIGH>;
+
+		buck1-supply = <&reg_vsodimm>;
+		buck2-supply = <&reg_vsodimm>;
+		buck3-supply = <&reg_vsodimm>;
+		ldo1-supply = <&reg_3v3>;
+		ldo2-supply = <&reg_1v8>;
+		ldo3-supply = <&reg_3v3>;
+		ldo4-supply = <&reg_3v3>;
+		system-power-controller;
+		ti,power-button;
+
 
 		regulators {
-			buck1_reg: buck1 {
+			reg_vdd_core: buck1 {
 				regulator-name = "VDD_CORE";
 				regulator-min-microvolt = <750000>;
-				regulator-max-microvolt = <750000>;
+				regulator-max-microvolt = <900000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			buck2_reg: buck2 {
+			reg_1v8: buck2 {
 				regulator-name = "VCC1V8";
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
@@ -579,7 +601,7 @@ buck2_reg: buck2 {
 				regulator-always-on;
 			};
 
-			buck3_reg: buck3 {
+			reg_vdd_ddr: buck3 {
 				regulator-name = "VDD_LPDDR4";
 				regulator-min-microvolt = <1100000>;
 				regulator-max-microvolt = <1100000>;
@@ -587,13 +609,16 @@ buck3_reg: buck3 {
 				regulator-always-on;
 			};
 
-			ldo1_reg: ldo1 {
+			reg_sd_3v3_1v8: ldo1 {
 				regulator-name = "VDDSHV_SD_IO_PMIC";
-				regulator-min-microvolt = <33000000>;
-				regulator-max-microvolt = <33000000>;
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-allow-bypass;
+				regulator-always-on;
+				regulator-boot-on;
 			};
 
-			ldo2_reg: ldo2 {
+			reg_vddr_core: ldo2 {
 				regulator-name = "VDDAR_CORE";
 				regulator-min-microvolt = <850000>;
 				regulator-max-microvolt = <850000>;
@@ -601,18 +626,18 @@ ldo2_reg: ldo2 {
 				regulator-always-on;
 			};
 
-			ldo3_reg: ldo3 {
+			reg_1v8a: ldo3 {
 				regulator-name = "VDDA_1V8";
-				regulator-min-microvolt = <18000000>;
-				regulator-max-microvolt = <18000000>;
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo4_reg: ldo4 {
+			reg_eth_2v5: ldo4 {
 				regulator-name = "VDD_PHY_2V5";
-				regulator-min-microvolt = <25000000>;
-				regulator-max-microvolt = <25000000>;
+				regulator-min-microvolt = <2500000>;
+				regulator-max-microvolt = <2500000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
@@ -630,12 +655,14 @@ &mcu_gpio0 {
 };
 
 &main_gpio0 {
+	status = "okay";
 	bootph-all;
 	pinctrl-names = "default";
 	pinctrl-0 = <&main_gpio0_digital_pins>;
 };
 
 &main_gpio1 {
+	status = "okay";
 	bootph-all;
 	pinctrl-names = "default";
 	pinctrl-0 = <&main_gpio1_digital_pins>;
@@ -697,6 +724,7 @@ &cpsw3g_mdio {
 	cpsw3g_phy0: ethernet-phy@0 {
 		bootph-all;
 		reg = <0>;
+		ti,clk-output-sel = <DP83867_CLK_O_SEL_OFF>;
 		ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_00_NS>;
 		ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_4_B_NIB>;
 	};
@@ -704,6 +732,7 @@ cpsw3g_phy0: ethernet-phy@0 {
 	cpsw3g_phy1: ethernet-phy@1 {
 		bootph-all;
 		reg = <1>;
+		ti,clk-output-sel = <DP83867_CLK_O_SEL_OFF>;
 		ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_00_NS>;
 		ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_4_B_NIB>;
 	};
@@ -721,11 +750,6 @@ mbox_main_r5fss0_core0: mbox-main-r5fss0-core0 {
 		ti,mbox-rx = <0 0 2>;
 		ti,mbox-tx = <1 0 2>;
 	};
-
-	mbox_main_r5fss0_core1: mbox-main-r5fss0-core1 {
-		ti,mbox-rx = <2 0 2>;
-		ti,mbox-tx = <3 0 2>;
-	};
 };
 
 &mailbox0_cluster4 {
@@ -741,6 +765,11 @@ mbox_m4_0: mbox-m4-0 {
 	};
 };
 
+/* We are using AM6412 which has single R5F core. Let's use Single-CPU mode */
+&main_r5fss0 {
+	ti,cluster-mode = <2>;	/* 0 = Split Mode, 2 = Single-CPU Mode */
+};
+
 &main_r5fss0_core0 {
 	mboxes = <&mailbox0_cluster2>, <&mbox_main_r5fss0_core0>;
 	memory-region = <&main_r5fss0_core0_dma_memory_region>,
@@ -748,21 +777,19 @@ &main_r5fss0_core0 {
 };
 
 &main_r5fss0_core1 {
-	mboxes = <&mailbox0_cluster2>, <&mbox_main_r5fss0_core1>;
-	memory-region = <&main_r5fss0_core1_dma_memory_region>,
-			<&main_r5fss0_core1_memory_region>;
+	status = "disabled";
 };
 
 &main_r5fss1 {
-    status = "disabled";
+	status = "disabled";
 };
 
 &main_r5fss1_core0 {
-    status = "disabled";
+	status = "disabled";
 };
 
 &main_r5fss1_core1 {
-    status = "disabled";
+	status = "disabled";
 };
 
 &serdes_ln_ctrl {
@@ -787,10 +814,26 @@ &pcie0_rc {
 	phys = <&serdes0_pcie_link>;
 	phy-names = "pcie-phy";
 	num-lanes = <1>;
+	pcie_aspm = "powersave";
 };
 
 &pcie0_ep {
+	status = "disabled";
 	phys = <&serdes0_pcie_link>;
 	phy-names = "pcie-phy";
 	num-lanes = <1>;
 };
+
+&cpu0 {
+	/* clocks = <&k3_clks 135 0>; */
+	clock-names = "cpu";
+	operating-points-v2 = <&a53_opp_table>;
+	cpu-supply = <&reg_vdd_core>;
+};
+
+&cpu1 {
+	/* clocks = <&k3_clks 136 0>; */
+	clock-names = "cpu";
+	operating-points-v2 = <&a53_opp_table>;
+	cpu-supply = <&reg_vdd_core>;
+};
\ No newline at end of file
