diff --git a/arch/arm64/boot/dts/ti/k3-am642-evm.dts b/arch/arm64/boot/dts/ti/k3-am642-evm.dts
index 91d726ef7594..0a9a2f648984 100644
--- a/arch/arm64/boot/dts/ti/k3-am642-evm.dts
+++ b/arch/arm64/boot/dts/ti/k3-am642-evm.dts
@@ -265,6 +265,16 @@ AM64X_IOPAD(0x0234, PIN_OUTPUT, 0) /* (C16) UART0_TXD */
 		>;
 	};
 
+	main_uart3_pins_default: main-uart3-default-pins {
+		bootph-all;
+		pinctrl-single,pins = <
+			AM64X_IOPAD(0x0060, PIN_INPUT, 2) /* (T17) UART3_CTSn */
+			AM64X_IOPAD(0x0174, PIN_OUTPUT, 10) /* (R3) UART3_RTSn */
+			AM64X_IOPAD(0x0248, PIN_INPUT, 4) /* (D16) UART3_RXD */
+			AM64X_IOPAD(0x024c, PIN_OUTPUT, 4) /* (E16) UART3_TXD */
+		>;
+	};
+
 	main_spi0_pins_default: main-spi0-default-pins {
 		pinctrl-single,pins = <
 			AM64X_IOPAD(0x0210, PIN_INPUT, 0) /* (D13) SPI0_CLK */
@@ -400,6 +410,14 @@ &main_uart1 {
 	pinctrl-0 = <&main_uart1_pins_default>;
 };
 
+&main_uart3 {
+	bootph-all;
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&main_uart3_pins_default>;
+	current-speed = <115200>;
+};
+
 &main_i2c0 {
 	bootph-all;
 	status = "okay";
