diff --git a/arch/arm64/boot/dts/perle/k3-am642-igos.dtsi b/arch/arm64/boot/dts/perle/k3-am642-igos.dtsi
index 98215d7..940af23 100644
--- a/arch/arm64/boot/dts/perle/k3-am642-igos.dtsi
+++ b/arch/arm64/boot/dts/perle/k3-am642-igos.dtsi
@@ -20,8 +20,8 @@
 
 	device-info {
 		compatible = "perle,device-info";
-		nvmem-cells = <&vendor>, <&serial>, <&uuid>, <&product>, <&model>, <&wifi_mac_addr0>, <&wifi_mac_addr1>, <&wifi_mac_addr2>, <&wifi_mac_addr3>, <&mac_addr2>, <&mac_addr3>, <&mac_addr4>, <&mac_addr5>, <&mac_addr6>, <&mac_addr7>, <&mac_addr8>;
-		nvmem-cell-names = "vendor", "serial", "uuid", "product", "model", "wifi_mac_addr0", "wifi_mac_addr1", "wifi_mac_addr2", "wifi_mac_addr3", "mac_addr2", "mac_addr3", "mac_addr4", "mac_addr5", "mac_addr6", "mac_addr7", "mac_addr8";
+		nvmem-cells = <&vendor>, <&serial>, <&uuid>, <&product>, <&model>, <&ssid>, <&password>, <&wifi_mac_addr0>, <&wifi_mac_addr1>, <&wifi_mac_addr2>, <&wifi_mac_addr3>, <&mac_addr2>, <&mac_addr3>, <&mac_addr4>, <&mac_addr5>, <&mac_addr6>, <&mac_addr7>, <&mac_addr8>;
+		nvmem-cell-names = "vendor", "serial", "uuid", "product", "model", "ssid", "password", "wifi_mac_addr0", "wifi_mac_addr1", "wifi_mac_addr2", "wifi_mac_addr3", "mac_addr2", "mac_addr3", "mac_addr4", "mac_addr5", "mac_addr6", "mac_addr7", "mac_addr8";
 	};
 
 	memory@80000000 {
@@ -492,6 +492,14 @@
 				reg = <0x30 0x10>;
 			};
 
+			ssid: ssid@40 {
+				reg = <0x40 0x10>;
+			};
+
+			password: password@50 {
+				reg = <0x50 0x10>;
+			};
+
 			uuid: uuid@60 {
 				reg = <0x60 0x10>;
 				perle,bytes;
