From b2e39db56640ba7d437a5edbd67889425969623e Mon Sep 17 00:00:00 2001
From: Kyrylo Yatsenko <hedrok@gmail.com>
Date: Sat, 30 Aug 2025 22:20:02 +0300
Subject: [PATCH 2/2] Fix lu -> llu where 64bit

---
 ipt_NETFLOW.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ipt_NETFLOW.c b/ipt_NETFLOW.c
index fefe809..08e9442 100644
--- a/ipt_NETFLOW.c
+++ b/ipt_NETFLOW.c
@@ -759,11 +759,11 @@ static int nf_seq_show(struct seq_file *seq, void *v)
 		    sampler_mode_string(),
 		    get_sampler_interval());
 		if (get_sampler_mode() != SAMPLER_HASH)
-			seq_printf(seq, " Flows selected %lu, discarded %lu.",
+			seq_printf(seq, " Flows selected %llu, discarded %llu.",
 			    atomic64_read(&flows_selected),
 			    atomic64_read(&flows_observed) - atomic64_read(&flows_selected));
 		else
-			seq_printf(seq, " Flows selected %lu.", atomic64_read(&flows_selected));
+			seq_printf(seq, " Flows selected %llu.", atomic64_read(&flows_selected));
 		seq_printf(seq, " Pkts selected %llu, discarded %llu.\n",
 		    t.pkts_selected,
 		    t.pkts_observed - t.pkts_selected);
-- 
2.50.1

